src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -309,14 +305,44 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 throw new ArgumentNullException("stream");
             }
 
-            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>();
+            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>(this);
             try
             {
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    Debug.Assert(_bufferedContent.Position == 0, "CopyToAsync assumes that _bufferedContent.Position == 0.");
+#if NETNative
```

Comment:
MemoryStream.TryGetBuffer is available in the 4.0.10 version of the System.IO contract.  System.Net.Http already references the 4.0.10 version.